pkg/segment/aggregations/segaggs.go

```diff
@@ -1937,14 +1939,60 @@ func performMultiValueColRequestWithoutGroupby(letColReq *structs.LetColumnsRequ
 		case "makemv":
 			finalValue := performMakeMV(fieldValueStr, mvColReq)
 			rec[mvColReq.ColName] = finalValue
+		case "mvexpand":
+			expandedRecs := performMVExpand(fieldValue)
+
+			// Apply limit if mvColReq.Limit is greater than 0
+			if mvColReq.Limit > 0 && len(expandedRecs) > int(mvColReq.Limit) {
+				expandedRecs = expandedRecs[:mvColReq.Limit]
+			}
+			delete(recs, key)
+			for i, expandedValue := range expandedRecs {
+				expandedValueStr, ok := expandedValue.(string)
+				if !ok {
+					expandedValueStr = fmt.Sprintf("%v", expandedValue)
+				}
+				newRec := make(map[string]interface{})
+				for k, v := range rec {
+					newRec[k] = v
+				}
+				newRec[mvColReq.ColName] = expandedValueStr
+				newRecs[fmt.Sprintf("%s_%d", key, i)] = newRec
+			}
 		default:
 			return fmt.Errorf("performMultiValueColRequestWithoutGroupby: unknown command %s", mvColReq.Command)
 		}
 	}
-
+	if mvColReq.Command == "mvexpand" {
+		for k, v := range newRecs {
+			recs[k] = v
+		}
+	}
 	return nil
 }
 
+func performMVExpand(fieldValue interface{}) []interface{} {
+	var values []interface{}
+
+	switch v := fieldValue.(type) {
+	case []string:
+		for _, val := range v {
+			values = append(values, val)
+		}
+	case []interface{}:
+		values = v
+	case string:
+		splitValues := strings.Split(v, ",")
+		for _, val := range splitValues {
+			values = append(values, val)
+		}
```

Comment:
I am missing something here? If it is a string, then it is not a Multi Value field right? 
Or is this how splunk does? if you give `mvexpand` on a string field?

Reply:
Handling just multi-value types should be enough here since we don't interpret comma separated strings as multi-value fields without using `makemv`